web-installer/setup-owncloud.php

```diff
@@ -32,42 +34,82 @@
 @set_time_limit(0);
  
 /**
- * @brief Setup class with a few helper functions
+ * Setup class with a few helper functions
  *
  */ 
 class oc_setup {
 
+	private static $requirements = array(
+		'classes' => array(
+			'ZipArchive' => 'zip',
+			'DOMDocument' => 'dom',
+		),
+		'functions' => array(
+			'xml_parser_create' => 'libxml',
+			'mb_detect_encoding' => 'mb multibyte',
+			'ctype_digit' => 'ctype',
+			'json_encode' => 'JSON',
+			'gd_info' => 'GD',
+			'gzencode' => 'zlib',
+			'iconv' => 'iconv',
+			'simplexml_load_string' => 'SimpleXML'
+		),
+		'defined' => array(
+			'PDO::ATTR_DRIVER_NAME' => 'PDO'
+		)
+	);
  
 	/**
-	* @brief Checks if all the ownCloud dependencies are installed
+	* Checks if all the ownCloud dependencies are installed
 	* @return string with error messages
 	*/ 
 	static public function checkdependencies() {
-		$error='';
-		
-		// do we have PHP 5.3.2 or newer?
-                if(version_compare(PHP_VERSION, '5.3.2', '<')) {
-			$error.='PHP 5.3.2 is required. Please ask your server administrator to update PHP to version 5.3.2 or higher. PHP 5.2 is no longer supported by ownCloud and the PHP community.';
+		$error = '';
+		$missingDependencies = array();;
+
+		// do we have PHP 5.3.3 or newer?
+		if(version_compare(PHP_VERSION, '5.3.3', '<')) {
+			$error.='PHP 5.3.3 is required. Please ask your server administrator to update PHP to version 5.3.3 or higher.<br/>';
```

Comment:
Hi


could you provide the instructions to compile owncloud client for Windows on sure 13?

Sent from Yahoo Mail on Android

From:"Lukas Reschke" <notifications@github.com>
Date:Wed, Feb 18, 2015 at 12:28 PM
Subject:Re: [administration] Stop execution in case the script is used for an upgrade (#11)

In web-installer/setup-owncloud.php:

> 	* @return string with error messages > 	*/ > 	static public function checkdependencies() { > -		$error=''; > -		 > -		// do we have PHP 5.3.2 or newer? > - if(version_compare(PHP_VERSION, '5.3.2', '<')) { > -			$error.='PHP 5.3.2 is required. Please ask your server administrator to update PHP to version 5.3.2 or higher. PHP 5.2 is no longer supported by ownCloud and the PHP community.'; > +		$error = ''; > +		$missingDependencies = array();; > + > +		// do we have PHP 5.3.3 or newer? > +		if(version_compare(PHP_VERSION, '5.3.3', '<')) { > +			$error.='PHP 5.3.3 is required. Please ask your server administrator to update PHP to version 5.3.3 or higher.<br/>'; 

We still ship 7.0 with the installer as far I know

—
Reply to this email directly or view it on GitHub.￼